project_timelog/models/project_timelog.py

```diff
@@ -1,57 +1,63 @@
 # -*- coding: utf-8 -*-
 import datetime
-import time
-from openerp import models, fields, api, SUPERUSER_ID
-from openerp.exceptions import Warning
-from openerp import http
+from openerp import models, fields, api
+from openerp.exceptions import Warning as UserError
+from openerp.tools.translate import _
 
 
-class ProjectTimeLog(models.Model):
+class ProjectTimelog(models.Model):
     _name = "project.timelog"
     _description = "project timelog"
+    _rec_name = 'work_id'
 
-    work_id = fields.Many2one("project.task.work", "Task", required=True)
+    work_id = fields.Many2one("project.task.work", "Task", required=True, index=True)
     task_name = fields.Char(related='work_id.task_id.name', store=True)
     project_name = fields.Char(related='work_id.task_id.project_id.name', store=True)
     start_datetime = fields.Datetime(string="Start date", default=datetime.datetime.now())
     end_datetime = fields.Datetime(string="End date")
     duration = fields.Float(string="Duration", compute="_compute_duration", store=True)
```

Comment:
add another field

     corrected_duration = fields.Float(string="Duration", compute="_compute_duration", store=True)		

It will allow to have computed duration and corrected duration (it will optimize sum_time function)